public/app/features/alerting/unified/components/contact-points/DuplicateMessageTemplate.tsx

```diff
@@ -1,42 +1,62 @@
 import { RouteChildrenProps } from 'react-router-dom';
 
+import { Alert, LoadingPlaceholder } from '@grafana/ui';
 import { EntityNotFound } from 'app/core/components/PageNotFound/EntityNotFound';
 
-import { useAlertmanagerConfig } from '../../hooks/useAlertmanagerConfig';
 import { useAlertmanager } from '../../state/AlertmanagerContext';
-import { DuplicateTemplateView } from '../receivers/DuplicateTemplateView';
+import { generateCopiedName } from '../../utils/duplicate';
+import { stringifyErrorLike } from '../../utils/misc';
+import { updateDefinesWithUniqueValue } from '../../utils/templates';
+import { TemplateForm } from '../receivers/TemplateForm';
+
+import { useGetNotificationTemplate, useNotificationTemplates } from './useNotificationTemplates';
 
 type Props = RouteChildrenProps<{ name: string }>;
 
-const NewMessageTemplate = ({ match }: Props) => {
+const DuplicateMessageTemplate = ({ match }: Props) => {
   const { selectedAlertmanager } = useAlertmanager();
-  const { data, isLoading, error } = useAlertmanagerConfig(selectedAlertmanager);
+  const templateUid = match?.params.name ? decodeURIComponent(match?.params.name) : undefined;
+
+  const {
+    currentData: template,
+    isLoading,
+    error,
+  } = useGetNotificationTemplate({ alertmanager: selectedAlertmanager ?? '', uid: templateUid ?? '' });
+
+  const {
+    currentData: templates,
+    isLoading: templatesLoading,
+    error: templatesError,
+  } = useNotificationTemplates({ alertmanager: selectedAlertmanager ?? '' });
+
+  if (!selectedAlertmanager) {
+    return <EntityNotFound entity="Alertmanager" />;
+  }
 
-  const name = match?.params.name;
-  if (!name) {
+  if (!templateUid) {
     return <EntityNotFound entity="Notification template" />;
   }
 
-  if (isLoading && !data) {
-    return 'loading...';
+  if (isLoading || templatesLoading) {
+    return <LoadingPlaceholder text="Loading notification template" />;
   }
 
-  // TODO decent error handling
-  if (error) {
-    return String(error);
+  if (error || templatesError || !template || !templates) {
+    return (
+      <Alert title="Error loading notification template" severity="error">
+        {stringifyErrorLike(error)}
+      </Alert>
+    );
   }
 
-  if (!data) {
-    return null;
-  }
+  const duplicatedName = generateCopiedName(template.title, templates?.map((t) => t.title) ?? []);
```

Comment:
I think that we could probably ditch the approach of trying to append ` (copy {n})`, and instead do something like ` (copy_${dateTime})`. What do others think? That would scale in general so we don't have to worry about pagination etc.

Reply:
This is a good point, I like the idea of skipping the fetch for all templates 👌. Actually this autogenerated name will be updated in the UI...so, this date will be probably replaced by a more readable name

Reply:
I'm not entirely sure how pagination plays into this since the function just looks to increment the last `n` by 1 🤔 

There _is_ a possibility that a template with that name would already exist but that seems like an edge-case we don't have to design for since the intention is for the user to update the name.